train.py

```diff
@@ -149,6 +154,9 @@ def __init__(self, args, log):
         self.save_dir = args.save_dir
         self.log = log
         self.visualize_predictions = args.visualize_predictions
+        self.discriminator = discriminator
+        self.num_domains = 20  # TODO: We should be able set this to a meaningful number based on the data
```

Comment:
Would we want this to be an argument instead of a hard-coded constant?

Reply:
eventually it would be an argument to the clustering algorithm. By the time we get to this point, it would be set.